packages/valory/skills/check_stop_trading_abci/behaviours.py

```diff
@@ -140,27 +140,30 @@ def is_staking_kpi_met(self) -> Generator[None, None, bool]:
             return True
         return False
 
-    def _compute_stop_trading(self) -> Generator:
+    def _compute_stop_trading(self) -> Generator[None, None, bool]:
         # This is a "hacky" way of getting required data initialized on
         # the Trader: On first period, the FSM needs to initialize some
         # data on the trading branch so that it is available in the
         # cross-period persistent keys.
         if self.is_first_period:
             self.context.logger.debug(f"{self.is_first_period=}")
+            yield  # ensures this is a generator
             return False
 
         self.context.logger.debug(f"{self.params.disable_trading=}")
         if self.params.disable_trading:
+            yield  # ensures this is a generator
```

Comment:
@Adamantios I have added this fix because of the pyright warnings I was getting.
1. Basically the if blocks that were returning True or False directly, essentially was mismatching from the expected return type which was Generator.
2. The return type was expanded from Just Generator to Generator[None, None, bool]:

> The return type `Generator[None, None, bool]` indicates that this function is a generator that yields `None` values and ultimately returns a `bool`. This type annotation is appropriate because:
> 
> - The function uses `yield` to pause execution and produce values.
> - The function's final return value is a boolean (`True` or `False`).
> 
> This ensures that the function's behavior and type signature are consistent with its intended use as a generator.

3. `_compute_stop_trading` method is expected to return a `Generator`, but in the case of the first period check, it returns a `False` directly. This is causing the type mismatch warning from **Pyright**.

To fix this, I ensured that `_compute_stop_trading` always returns a `Generator`. I achieved this by using `yield` in the first period check and disable trading check  as well. 

basically the functionality is still the same but the code aligns much better with return type